core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -683,7 +683,8 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
    * Repeats this effect forever (until the first error). For more sophisticated
    * schedules, see the `repeat` method.
    */
-  final def forever: ZIO[R, E, Nothing] = self *> self.forever
+  final def forever: ZIO[R, E, Nothing] =
+    self *> ZIO.yieldNow *> forever
```

Comment:
These look awesome. Making ZIO "single-thread safe" by default is going to be amazing.